packages/sources/uniswap-v3/src/endpoint/crypto.ts

```diff
@@ -0,0 +1,151 @@
+import { Requester, Validator } from '@chainlink/ea-bootstrap'
+import { ExecuteWithConfig, InputParameters } from '@chainlink/types'
+import { NAME as AdapterName, Config } from '../config'
+import { ethers, BigNumber } from 'ethers'
+import quoterABI from '../abis/quoter.json'
+import erc20ABI from '../abis/ERC20.json'
+import { Decimal } from 'decimal.js'
+
+export const supportedEndpoints = ['crypto']
+
+export const endpointResultPaths = {
+  crypto: 'rate',
+}
+
+export interface ResponseSchema {
+  input: string
+  inputToken: string
+  inputDecimals: number
+  output: string
+  outputToken: string
+  outputDecimals: number
+  rate: number
+}
+
+export const inputParameters: InputParameters = {
+  from: ['base', 'from', 'coin'],
+  fromAddress: false,
+  fromDecimals: false,
+  to: ['quote', 'to', 'market'],
+  toAddress: false,
+  toDecimals: false,
+  amount: false,
+  resultPath: false,
+  feeTiers: false,
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+
+  if (validator.error) throw validator.error
+  const jobRunID = validator.validated.id
+  const { address: from, decimals: fromDecimals } = await getTokenDetails(validator, 'from', config)
+  const { address: to, decimals: toDecimals } = await getTokenDetails(validator, 'to', config)
+  const inputAmount = validator.validated.data.amount || 1
+  const amount = BigNumber.from(inputAmount).mul(BigNumber.from(10).pow(fromDecimals))
+  const resultPath = validator.validated.data.resultPath
+
+  const feeTiers = validator.validated.data.feeTiers || config.feeTiers
+  const output = await getBestRate(from, to, amount, feeTiers, config)
```

Comment:
Can we throw an error if the output is 0? Or in some way catch it if no pool exists for the from/to (or the requested fee tiers do not exists)?

Basically, we want to avoid returning the price as 0 if the pool/fee tiers do not exist.